fakelibusb_test.go

```diff
@@ -203,8 +211,9 @@ func newFakeLibusb() (*fakeLibusb, func() error) {
 		// libusbDevice only as an identifier, use an arbitrary unique pointer.
 		// The contents of these pointers is never accessed.
 		fl.fakeDevices[newDevicePointer()] = &fakeDevice{
```

Comment:
here if the same fakeDevice struct is reused for definitions, we could just do the copy of the whole struct.
As in:
d := d  // this would be necessary because through all iterations the for variable shares the same address
fl.fakeDevices[newDevicePointer()] = &d

Or alternatively, which would perhaps be more readable:
fd := new(fakeDevice)
*fd = d
fl.fakeDevices[devPtr] = fd

Reply:
done